orderbook/src/bad_token/list_based.rs

```diff
@@ -0,0 +1,116 @@
+use super::{BadTokenDetecting, TokenQuality};
+use anyhow::Result;
+use primitive_types::H160;
+
+/// If a token is neither in the allow nor the deny list treat it this way.
+pub enum UnknownTokenStrategy {
+    Allow,
+    Deny,
+    Forward(Box<dyn BadTokenDetecting>),
+}
+
+/// Classify tokens with explicit allow and deny lists.
+pub struct ListBasedDetector {
+    pub allow_list: Vec<H160>,
+    pub deny_list: Vec<H160>,
+    pub strategy: UnknownTokenStrategy,
+}
+
+#[async_trait::async_trait]
+impl BadTokenDetecting for ListBasedDetector {
+    async fn detect(&self, token: ethcontract::H160) -> Result<TokenQuality> {
+        if self.allow_list.contains(&token) {
+            return Ok(TokenQuality::Good);
```

Comment:
I guess we have to be careful not to include any actually bad tokens into the allow list since they essentially default to good.